lib/src/protocol/kawa_h1/answers.rs

```diff
@@ -1,100 +1,904 @@
-use std::{collections::HashMap, rc::Rc};
+use crate::{protocol::http::DefaultAnswer, sozu_command::state::ClusterId};
+use kawa::{
+    h1::NoCallbacks, AsBuffer, Block, BodySize, Buffer, Chunk, Kawa, Kind, Pair, ParsingPhase,
+    ParsingPhaseMarker, StatusLine, Store,
+};
+use sozu_command::proto::command::CustomHttpAnswers;
+use std::{
+    collections::{HashMap, VecDeque},
+    fmt,
+    rc::Rc,
+};
 
-use crate::{protocol::http::DefaultAnswerStatus, sozu_command::state::ClusterId};
+#[derive(Clone)]
+pub struct SharedBuffer(Rc<[u8]>);
 
-#[allow(non_snake_case)]
-pub struct DefaultAnswers {
-    /// 400
-    pub BadRequest: Rc<Vec<u8>>,
-    /// 401
-    pub Unauthorized: Rc<Vec<u8>>,
-    /// 404
-    pub NotFound: Rc<Vec<u8>>,
-    /// 408
-    pub RequestTimeout: Rc<Vec<u8>>,
-    /// 413
-    pub PayloadTooLarge: Rc<Vec<u8>>,
-    /// 502
-    pub BadGateway: Rc<Vec<u8>>,
-    /// 503
-    pub ServiceUnavailable: Rc<Vec<u8>>,
-    /// 504
-    pub GatewayTimeout: Rc<Vec<u8>>,
-    /// 507
-    pub InsufficientStorage: Rc<Vec<u8>>,
+impl AsBuffer for SharedBuffer {
```

Comment:
@Wonshtrum should we export a `AsReadOnlyBuffer` trait from kawa to remove the `panic` in implementation